controller/lib/pid/pid.cpp

```diff
@@ -23,21 +23,14 @@ float PID::Compute(Time now, float input, float setpoint) {
   if (!initialized_) {
     last_input_ = input;
     last_error_ = setpoint - input;
-    next_sample_time_ = now;
+
     // last call time defined as now - SampleTime to enable computation on first
     // call (user should call Compute() immediately after SetMode(Auto))
-    last_update_time_ = now - sample_period_;
     initialized_ = true;
   }
 
-  // compute actual samples time-difference to take jitter into account in
-  // integral and derivative
-  Duration effectiveSampleTime = (now - last_update_time_);
-  float samplesTimeChangeSec = effectiveSampleTime.seconds();
-  // condition to update output : 1 sample time has passed and we have new data
-  if (now < next_sample_time_ || samplesTimeChangeSec <= 0) {
-    return last_output_;
-  }
+  float samplesTimeChangeSec = sample_period_.seconds();
```

Comment:
This change means that we no longer need to account for jitter, because it happens on a very precise timer, correct? (and this is also why it's safe to delete the jitter tests without introducing new tests)

Reply:
Yes, the only remaining jitter will be caused by disabled interrupts (should be short and rare), or the time spent in interrupt handlers.  Probably in the microsecond region.